pyriemann/regression.py

```diff
@@ -0,0 +1,174 @@
+        y : ndarray, shape (n_matrices, 1)
+            labels corresponding to each matrix.
+
+        Returns
+        -------
+        self : Riemannian SVC instance
+            The SVC instance.
+        """
+        kernelmat = kernel(X, Cref=self.Cref, metric=self.metric)
+        self.data_ = X
+        self.svr_ = self.svr_.fit(kernelmat, y)
+
+        return self
+
+    def predict(self, X):
+        """Get the predictions.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+
+        Returns
+        -------
+        pred : ndarray of int, shape (n_matrices, 1)
+            Predictions for each matrix according to the SVC.
+        """
+        test_kernel_mat = kernel(X,
+                                 self.data_,
+                                 Cref=self.Cref,
+                                 metric=self.metric)
+        return self.svr_.predict(test_kernel_mat)
+
+
+class KNNRegression(MDM):
+    """Regression by K-Nearest-Neighbors.
+
+    Reression by nearest Neighbors. For each point of the test set, the
+    pairwise distance to each element of the training set is estimated. The
+    value is calculated according to the softmax average w.r.t. distance of
+    the k nearest neighbors.
+
+    Parameters
+    ----------
+    n_neighbors : int, (default: 5)
+        Number of neighbors.
+    metric : string | dict (default: 'riemann')
+        The type of metric used for distance estimation.
+        see `distance` for the list of supported metric.
+
+    Attributes
+    ----------
+    classes_ : list
+        list of classes.
+
+    """
+
+    def __init__(self, n_neighbors=5, metric='riemann'):
+        """Init."""
+        # store params for cloning purpose
+        self.n_neighbors = n_neighbors
+        MDM.__init__(self, metric=metric)
+
+    def fit(self, X, y):
+        """Fit (store the training data).
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+        y : ndarray shape (n_matrices, 1)
+            labels corresponding to each matrix.
+
+        Returns
+        -------
+        self : NearestNeighbor instance
+            The NearestNeighbor instance.
+        """
+        self.values_ = y
+        self.covmeans_ = X
+
+        return self
+
+    def predict(self, covtest):
+        """Get the predictions.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+
+        Returns
+        -------
+        pred : ndarray of int, shape (n_matrices, 1)
+            Predictions for each matrix according to the closest centroid.
+        """
+        dist = self._predict_distances(covtest)
+        dist_sorted = np.sort(dist)
+        neighbors_values = self.values_[np.argsort(dist)]
+        softmax_dist = softmax(-dist_sorted[:, 0:self.n_neighbors])
```

Comment:
IMO, it should be the softmax of squared distances (see #100):
```suggestion
        softmax_dist = softmax(-dist_sorted[:, 0:self.n_neighbors]**2)
```